app.py

```diff
@@ -1,33 +1,54 @@
 from flask import Flask, render_template, request, redirect, url_for
+from flask_sqlalchemy import SQLAlchemy
 
 app = Flask(__name__)
+app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite:///pantry.db'
+app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False
+db = SQLAlchemy(app)
 
-# A list to store pantry items (for demonstration purposes)
-pantry_items = []
+class PantryItem(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    item = db.Column(db.String(100), nullable=False)
+    quantity = db.Column(db.Integer, nullable=False)
+
+    def __repr__(self):
+        return f'<PantryItem {self.item}>'
+
+# Initialize the database
+with app.app_context():
+    # Delete existing database tables
+    db.drop_all()
+    # Create table
+    db.create_all()
 
 @app.route('/')
 def home():
     return render_template('home.html')
 
+# Adding item to database
 @app.route('/inventory', methods=['GET', 'POST'])
 def inventory():
     if request.method == 'POST':
         item = request.form['item']
         quantity = request.form['quantity']
-        pantry_items.append({'item': item, 'quantity': int(quantity)})
+        new_item = PantryItem(item=item, quantity=int(quantity))
+        db.session.add(new_item)
+        db.session.commit()
         return redirect(url_for('inventory'))
+    pantry_items = PantryItem.query.all()
     return render_template('inventory.html', pantry_items=pantry_items)
 
-# needs work to become fully functional
+# Editing item in database
 @app.route('/edit', methods=['GET', 'POST'])
 def edit():
     if request.method == 'POST':
-        item = request.form['item']
+        item_id = request.form['item_id']
         quantity = request.form['quantity']
-        for entry in pantry_items:
-            if entry.get(item):
-                entry[item] = quantity
+        item = PantryItem.query.get(item_id)
```

Comment:
I wouldn't worry about its interactions with the front-end as that is what I am currently implementing. It would be best to place all function relative to models in its own separate file